src/lib/EventPublisher.js

```diff
@@ -59,6 +74,11 @@ class EventPublisher extends EventEmitter {
 
     this._emitWithPrefix(eventData.name, event);
     this.emit(ALL_EVENTS, event);
+
+    // broadcast events only from the current worker
+    if (!event.fromMaster && cluster.isWorker) {
```

Comment:
I think that instead of having a `fromMaster` property you can just refactor the `publish` function.

You would have `(process: any).send(event)` in `publish` and move the rest of the code to `_publish`. Then `process.on('message', (event: Event) => this._publish(event))` and I think you wouldn't need to worry about `fromMaster`